src/k8s-extension/azext_k8s_extension/partner_extensions/Dapr.py

```diff
@@ -4,43 +4,153 @@
+        if prompt_y_n(self.MSG_IS_DAPR_INSTALLED, default='n'):
+            dapr_exists = True
+
+            name = prompt(self.MSG_ENTER_RELEASE_NAME, self.RELEASE_INFO_HELP_STRING) or self.DEFAULT_RELEASE_NAME
+            if release_name and name != release_name:
+                logger.info("The release name has been changed from '%s' to '%s'.", release_name, name)
+
+            namespace = prompt(self.MSG_ENTER_RELEASE_NAMESPACE, self.RELEASE_INFO_HELP_STRING)\
+                or self.DEFAULT_RELEASE_NAMESPACE
+            if release_namespace and namespace != release_namespace:
+                logger.info("The release namespace has been changed from '%s' to '%s'.", release_namespace, namespace)
+
+        return name, namespace, dapr_exists
+
+    def _reset_configuration_settings_for_existing_dapr(self, configuration_settings: dict) -> Dict:
+        '''
+        Reset some configuration settings for the extension if Dapr is already installed in the cluster.
+        This is required because some settings modify the StatefulSet and is unsupported by K8s.
+        '''
+        for key in configuration_settings.keys():
+            if key.startswith(self.HA_KEY_PREFIX) or key.startswith(self.PLACEMENT_KEY_PREFIX):
+                logger.warning("The configuration setting '%s' will be ignored as Dapr is already installed, "
+                               "please refer to %s for more information.", key, self.TSG_LINK)
+                del configuration_settings[key]
+        return configuration_settings
+
+    def Create(self, cmd, client, resource_group_name: str, cluster_name: str, name: str, cluster_type: str,
+               cluster_rp: str, extension_type: str, scope: str, auto_upgrade_minor_version: bool,
+               release_train: str, version: str, target_namespace: str, release_namespace: str,
+               configuration_settings: dict, configuration_protected_settings: dict,
+               configuration_settings_file: str, configuration_protected_settings_file: str):
         """ExtensionType 'Microsoft.Dapr' specific validations & defaults for Create
            Must create and return a valid 'ExtensionInstance' object.
         """
 
-        if cluster_type.lower() == '' or cluster_type.lower() == 'managedclusters':
-            configuration_settings[self.CLUSTER_TYPE] = 'managedclusters'
+        # Dapr extension is only supported at the cluster scope.
+        if scope == 'namespace':
+            raise InvalidArgumentValueError(self.ERR_MSG_INVALID_SCOPE_TPL.format(scope))
+
+        release_name, release_namespace, dapr_exists = \
+            self._get_release_info(name, release_namespace, configuration_settings)
+
+        # Inform the user that the extension will be installed on an existing Dapr installation.
+        # Reset some configuration settings for the extension if Dapr is already installed in the cluster.
+        if dapr_exists:
```

Comment:
Couple of questions/clarifications here - 
1) if OSS version installed is say 1.8 and someone is trying to install dapr thru extension but version is 1.9 then in that case whether `create` will upgrade all the control plane?
2) We are not actually uninstalling the oss dapr here, how create is upgrading the charts for the existing cluster ?

Reply:
1. Helm runs an upgrade, so yes, control plane will be upgraded to 1.9
1. If the release name and namespace matches, helm upgrades the charts with the newest configuration and the extension starts pointing to it
1. It's the same as issuing a `helm upgrade` on your own, and Dapr supports zero-downtime upgrades, so there is no downtime